babble-backend/routes-documentation.txt

```diff
@@ -0,0 +1,78 @@
+Notes on the routes
+
+**************************************** need email/pasword in body ****************************************
+
+Pass in email/password (key-value pairs in json format) in the BODY
+
+POST /enter/login
+        - pass in email/password (see top of section)
+        - get back: auth-token (store this)
+
+POST /enter/signup
+        - pass in email/password (see top of section)
+
+******************************** need auth-token (get from login) in header ********************************
+
+Pass in auth-token (get it from login) as a key-value pair like so ----> Authentication: <the token>
+in json format in the HEADER
+
+POST /user/change-password
+        - pass in auth-token (see top of section)
+        - req.body: email, oldPassword, newPassword (key-value json format)
+        - call logout route immediately after (if successful change)
+        - ask user to log back in
+
+POST /user/change-email
+        - pass in auth-token (see top of section)
+        - req.body: oldEmail, newEmail, password (key-value json format)
+        - call logout route immediately after (if successful change)
+        - ask user to log back in
+
+DELETE /user/delete-account
+        - pass in auth-token (see top of section)
+        - req.body: email, password (key-value json format)
+        - call logout route immediately after (if successful change)
+
+POST /user/saved-posts
+        - pass in auth-token (see top of section)
+        - get back: array of post objects
+
+GET /user/my-posts
+        - pass in auth-token (see top of section)
+        - get back: array of post objects
+
+DELETE /user/delete-post/:pid   
+        - pass in auth-token (see top of section)
+        - query: post id (pid)
+
+GET /user/logout
+        - pass in auth-token (see top of section)
+
+GET /posts/feed
+        - pass in auth-token (see top of section)
+
+POST /posts/create-post 
+        - pass in auth-token (see top of section)
+        - req.body: title, body, tags (array of tags)
+
+POST /posts/comment/:pid
+        - pass in auth-token (see top of section)
+        - req.body: comment
```

Comment:
```suggestion
        - req.body: comment body, poster name
        - req.params: post id (pid)
```

Reply:
I know that the documentation says to allow users to pick a name when posting, but doesn't that defeat the point of an anonymous app? Like what if I always posted as RupinMittal123? For the current setup, this change looks good, but I think we should remove this feature of picking names to post under. Any thoughts?

Reply:
Thats really up to the user, we want to provide the ability to be anonymous, but you can change your name freely at any time, so it stays anonymous as much as you want.

Reply:
Alright, fair point. Now that you mention it, yeah, it's better to give them the option rather than forcing it. This change looks good then.